src/pool-priority-sort.js

```diff
@@ -0,0 +1,13 @@
+export default function poolPrioritySort(a, b) {
+  if (a.connection.connected && !b.connection.connected) {
+    return -1;
+  }
+  if (!a.connection.connected && b.connection.connected) {
+    return 1;
+  }
+  if (!a.connection.connected && !b.connection.connected) {
```

Comment:
It is more conventional to not have this compare and return.  We would also get the benefit of sorting the disconnected pools in priority.